src/shared/dns-domain.c

```diff
@@ -103,6 +107,10 @@ int dns_label_unescape(const char **name, char *dest, size_t sz) {
 
                         /* Normal character */
 
+                        if (FLAGS_SET(flags, DNS_LABEL_LDH) && r == 0 && *n == '-')
+                                /* Leading dash */
+                                return -EINVAL;
+
```

Comment:
Hmm, this current function is happy to accept non-7bit chars too (it has too, mdns needs that). I figure when the flag is set we should refuse that too, in fact following the excerpt of the RFC you quoted we should probably refuse a lot more even, including non-alphanumerics and stuff

Reply:
(that said, does glibc honour escapes in /etc/hosts? if not, then we probably should not honour these either here. But I think that should be a separate flag then, i.e. DNS_LABEL_NO_ESCAPES or so, that the /etc/hosts parser then sets.

Reply:
It doesn't:
```console
$ grep abc /etc/hosts                                                                                            
1.2.4.5  a\097a abc                                                                                              
$ getent hosts aaa                                                                                               
$ getent hosts $'a\\097a'                                                                                        
1.2.4.5         a\097a abc                                                                                       
```